### PR TITLE
Week 2 assessment ja

### PR DESCRIPTION
**Pull request by joagui83**
Completed the assessment.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,56 +14,224 @@
...
-const num2 = 0
-// Expected output: "0 is divisible by three"
-const num3 = -7
-// Expected output: "-7 is not divisible by three"
+describe("divisibletBy3", () => {
+    it(" takes in a number and returns number is divisible by three", () => {
+    const num1= 15
+    const num2 = 0
+    const num3 = -7
+    const divBy3 = (number) => {
```
Comment (elysemontano) [inline code-challenges.test.js:59]: Tests are not meant to contain any logic.  They are simply there to take an input, run it through your code and check if the output is what you expect.  While this test works because you have all of the necessary things to run a successful test, lines 59 through 63 are not needed inside your test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,56 +14,224 @@
...
+ const num1 = 15
+// // Expected output: "15 is divisible by three"
+ const num2 = 0
+// // Expected output: "0 is divisible by three"
+ const num3 = -7
+// // Expected output: "-7 is not divisible by three"
+const divBy3 = (number) => {
+    if(number % 3 == 0){
+        return `${number} is divisible by three`
+    }else{ return `${number} is not divisible by three`}
```
Comment (elysemontano) [inline code-challenges.test.js:99]: Excellent, great job on this!  Oftentimes, a return will be on the following line for readability.

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,56 +14,224 @@
...
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.097 s, estimated 1 s
 
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
+describe("pascalCase", () => {
```
Comment (elysemontano) [inline code-challenges.test.js:118]: Awesome name for the function!

File: code-challenges.test.js [hunk 4]
```diff
@@ -14,56 +14,224 @@
...
+    })
+  })
+
+//   FAIL  ./code-challenges.test.js
+//   pascalCase
+//     ✕ takes an array of strings and returns an array with the first index of all letters capitilized
+
+//   ● pascalCase › takes an array of strings and returns an array with the first index of all letters capitilized
+
+//     ReferenceError: pascalCase is not defined
```
Comment (elysemontano) [inline code-challenges.test.js:134]: Test looks great!

File: code-challenges.test.js [hunk 5]
```diff
@@ -14,56 +14,224 @@
...
+ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+ const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
+// // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
+
+const pascalCase = (input) => {
+    return input.map(element => {
+        return element.charAt(0).toUpperCase() + element.substring(1).toLowerCase();}
+
+     )}
```
Comment (elysemontano) [inline code-challenges.test.js:153]: Awesome!  Great job in keeping this really thorough by lowercasing the substring as well.  This is definitely a good practice to think of possible edge cases and address them early on.

File: code-challenges.test.js [hunk 6]
```diff
@@ -14,56 +14,224 @@
...
+    it("takes in a string and logs the index of the first vowel", () => {
+        const vowelTester1 = "learn"
+        const vowelTester2 = "academy"
+        const vowelTester3 = "challenges"
+
+    expect(findFirstVowel( vowelTester1)).toEqual(1)
+    expect(findFirstVowel(vowelTester2)).toEqual(0)
+    expect(findFirstVowel(vowelTester3)).toEqual(2)
+})
+})
```
Comment (elysemontano) [inline code-challenges.test.js:193]: Another great test!  Just make sure to watch the indentation.  If you have two curlies that are directly on top of each other, oftentimes there may be some indentation off somewhere.

File: code-challenges.test.js [hunk 7]
```diff
@@ -14,56 +14,224 @@
...
+// forLoop to loop through the string 
+
+function findFirstVowel(string){
+    let vowels = 'aeiou'
+    for(let i =0; i<string.length; i++){
+      if(vowels.includes(string[i])){
+        return i
+      }
+    }
+  }
```
Comment (elysemontano) [inline code-challenges.test.js:225]: Function looks great, would love to hear your thoughts on why you chose function notation versus arrow notation for this problem.

File: code-challenges.test.js [hunk 8]
```diff
@@ -14,56 +14,224 @@
...
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
+console.log(findFirstVowel(vowelTester1));
 
+console.log(findFirstVowel(vowelTester2));
 
-// b) Create the function that makes the test pass.
+console.log(findFirstVowel(vowelTester3))
```
Comment (elysemontano) [inline code-challenges.test.js:237]: No need for console logs or function invocation since this all is happening from your tests.

File: coding-concepts.js [hunk 9]
```diff
@@ -67,7 +68,31 @@ class Learn {
   }
 }
 var learnStudent = new Learn("George")
-// console.log(learnStudent)
+ console.log(learnStudent)
+
+// a) Your answer:{name: "George", cohort: "Bravo", year:"2022"}
+// b) Verify and explain: {name: "George", cohort: "Bravo", year:"2022"}
+//learnStudent is a child class with the instance of name: George created but it inherits the rest of the dat from the parent class: Learn since they are static and not dynamic like name is.
```
Comment (elysemontano) [inline coding-concepts.js:75]: In this case, there is no child class.  A child class would be defined as `class Child extends Parent`.  In this case, new Learn on line 70 is creating a new instance of an object from the Learn class and assigning that object to the variable learnStudent.

File: interview-questions.md [hunk 10]
```diff
@@ -8,49 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: Parameter is what you set call back for your variable data types.
+  Arguments reference the paramaters when you do your callbacks. Arguments must match the paramaters set in the variables
 
-  Researched answer:
+  Researched answer:Parameters are named variabless that get passed into a function and are used to import arguments into a function.
+  Argument are the values used to access the parameters passed into a function.
```
Comment (elysemontano) [inline interview-questions.md:15]: An argument is the data being passed into the function when invoking the function.  The parameter renames the data to a more generic name so that the code can be used dynamically and with any input.

File: interview-questions.md [hunk 11]
```diff
@@ -8,49 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: .map() can take the index and the original array  as a needed parmater. Index is an optional parameter of .map().
 
+  Researched answer: .map() parametera are the definition of a function, Index, and callback of the original array. Index is an optional parameter, while function defition and callback of original array are needed parameters.
```
Comment (elysemontano) [inline interview-questions.md:23]: More specifically, the value is required when using .map().  Since .map is meant to iterate, map needs to iterate upon specific values and then do something to them.

File: interview-questions.md [hunk 12]
```diff
@@ -8,49 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer:Function is code that you can re use multiple times until it meets a specifig task
+  Method is a function within an objec
 
-  Researched answer:
+  Researched answer:function is  a series of code block machines designed to perform a particular task or calculates a value. 
+  Method
```
Comment (elysemontano) [inline interview-questions.md:42]: What was your researched answer on methods?

**Review by elysemontano — APPROVED** [review]
Great job Jose!  Would like to see a little more breakdown of how you are using the built-in methods in your pseudocode, but overall great work!  I left you some comments to clarify a few things in the interview questions and code challenges.

File: interview-questions.md (hunk 12 above)
Comment (joagui83) [reply interview-questions.md:42]: Method is a function within an object

File: interview-questions.md (hunk 11 above)
Comment (joagui83) [reply interview-questions.md:23]: Thank you I def was overthinking this one.

File: interview-questions.md (hunk 10 above)
Comment (joagui83) [reply interview-questions.md:15]: Got it!

File: code-challenges.test.js (hunk 7 above)
Comment (joagui83) [reply code-challenges.test.js:225]: It was by accident I re wrote the function into an arrow notation but I had already pushed up my code without it. Did not know function notation was a thing and that it worked with my test. 

let findFirstVowel = (string) => {
    let vowels = "aeiou"
    for(let i = 0; i < string.length; i++){
      if(vowels.includes(string[i])){
        return i 
      }
    }
  }
